servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -2956,6 +2957,38 @@ public DomainRoleMembers getDomainRoleMembers(ResourceContext ctx, String domain
         return roleMembers;
     }
 
+    @Override
+    public DomainRoleMember getPrincipalRoles(ResourceContext context, String principal, String domainName) {
+        final String caller = "getPrincipalRoles";
+        metric.increment(ZMSConsts.HTTP_GET);
+        logPrincipal(context);
+
+        if (StringUtil.isEmpty(principal)) {
+            // If principal not specified, get roles for current user
+            principal = ((RsrcCtxWrapper) context).principal().getFullName();
+        }
+        validateRequest(context.request(), caller);
+        validate(principal, TYPE_ENTITY_NAME, caller);
+
+        if (!StringUtil.isEmpty(domainName)) {
+            validate(domainName, TYPE_DOMAIN_NAME, caller);
```

Comment:
we should also lowercase the domainname here

Reply:
Fixed